sycl/source/detail/scheduler/circular_buffer_extended.cpp

```diff
@@ -0,0 +1,148 @@
+//==-- circular_buffer_extended.hpp - Circular buffer with host accessor ---==//
```

Comment:
Are we sure "Circular Buffer Extended" is the best name for this class? SYCL has `buffer` and we have `buffer_impl` (inheriting from `MemObjT`) underneath them.   Using `Buffer` in the name of this class suggests that it is another type of SYCL data buffer, which it is not.   
I'm not good at naming things, but maybe `CircularQueue`, `CircularMemQueue` or `CircularMemCollection` ?  
Sorry to inject a comment so late in the review cycle.

Reply:
> Using `Buffer` in the name of this class suggests that it is another type of SYCL data buffer, which it is not.   

It's not exposed to the user, so it won't confuse them.
I see the point of your confusion here. `CircularBufferExtended` says that this class extends `CircularBuffer`'s functionality. Although, it doesn't behave as `CircularBuffer` sometimes at all. On the other hand, the intent behind the class is to store leaves of graph. Hence, I think, something like `LeavesCollection` or `Leaves` would suite.

@sergey-semenov Are you OK with naming of the class?

Reply:
I'm fine with renaming it to `LeavesCollection`, since it's heavily customized for that purpose anyway.

Reply:
Done,